ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouterMetrics.java

```diff
@@ -277,6 +297,20 @@ public Integer getValue() {
   }
 
   /**
+   * Initializes a {@link Gauge} metric for the maxPutChunkSize. This value is a constant for a running router
+   * instance, but can change between invocations. This metric helps maintain historical information.
+   * @param maxPutChunkSize the value to initialize this metric with.
+   */
+  public void initializeMaxPutChunkSizeMetric(final int maxPutChunkSize) {
```

Comment:
Why is this required. We will have history to our config files right and I don't think this might change frequently either.